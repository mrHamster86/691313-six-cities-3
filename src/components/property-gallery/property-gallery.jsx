import React, {memo} from 'react';
import PropTypes from 'prop-types';

export const PropertyGallery = ({images, title}) => {
  const MAX_IMAGES = 6;
  images.length = images.length > MAX_IMAGES
    ? MAX_IMAGES
    : images.length;

  return (
    <div className="property__gallery">
      {images.map((img, index) => (
        <div key={index} className="property__image-wrapper">
          <img
            className="property__image"
            src={img}
            alt={title}
          />
        </div>
      ))}
    </div>
  );
};

PropertyGallery.propTypes = {
  images: PropTypes.arrayOf(PropTypes.string).isRequired,
  title: PropTypes.string.isRequired,
};

export default memo(PropertyGallery);
