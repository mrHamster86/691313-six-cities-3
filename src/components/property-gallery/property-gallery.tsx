import * as React from 'react';

interface Props {
  images: string[];
  title: string;
}

export const PropertyGallery: React.FC<Props> = ({images, title}: Props) => {
  const MAX_IMAGES = 6;
  images.length = images.length > MAX_IMAGES
    ? MAX_IMAGES
    : images.length;

  return (
    <div className="property__gallery">
      {images.map((img, index) => (
        <div key={index} className="property__image-wrapper">
          <img
            className="property__image"
            src={img}
            alt={title}
          />
        </div>
      ))}
    </div>
  );
};

export default React.memo(PropertyGallery);
