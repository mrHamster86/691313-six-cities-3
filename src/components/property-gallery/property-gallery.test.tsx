import * as React from 'react';
import * as renderer from 'react-test-renderer';
import {PropertyGallery} from './property-gallery';

const images = [`img/apartment-1.jpg`];
const title = `Canal View Prinsengracht`;

it(`PropertyGallery components renders`, () => {
  const tree = renderer.create(
      <PropertyGallery images={images} title={title}/>).toJSON();
  expect(tree).toMatchSnapshot();
});
