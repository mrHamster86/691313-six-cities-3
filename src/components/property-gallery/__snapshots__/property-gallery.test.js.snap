// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyGallery components renders 1`] = `
<div
  className="property__gallery"
>
  <div
    className="property__image-wrapper"
  >
    <img
      alt="Canal View Prinsengracht"
      className="property__image"
      src="img/apartment-1.jpg"
    />
  </div>
</div>
`;
